### PR TITLE
[RLlib][Docs] Restructure Utils API page

### PR DESCRIPTION
**Pull request by kouroshHakha**
## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: doc/source/rllib/package_ref/utils.rst [hunk 1]
```diff
@@ -3,14 +3,203 @@
...
-   utils/tf_utils.rst
-   utils/torch_utils.rst
-   utils/numpy.rst
-   utils/deprecation.rst
+Here is a list of all the utilities available in RLlib.
+
+Exploration API
+---------------
+
+Exploration in RL is crucial for a learning agent in order to more easily
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:11]:
```suggestion
Exploration is crucial in RL for enabling a learning agent to find new, potentially high-reward states by reaching unexplored areas of the environment.
```

File: doc/source/rllib/package_ref/utils.rst [hunk 2]
```diff
@@ -3,14 +3,203 @@
...
+reach areas of the environment that have not been discovered so far and therefore
+find new states yielding possibly high rewards.
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:13]:
```suggestion

```

File: doc/source/rllib/package_ref/utils.rst [hunk 3]
```diff
@@ -3,14 +3,203 @@
...
+
+RLlib comes with several built-in exploration components, used by
+the different algorithms. Also users can customize an algo's exploration
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:15]:
```suggestion
the different algorithms use. Users can also customize an algorithm's exploration
```

File: doc/source/rllib/package_ref/utils.rst [hunk 4]
```diff
@@ -3,14 +3,203 @@
...
+behavior by sub-classing the Exploration base class and implementing
+their own logic:
+
+Some Built-in Exploration Components
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:18]:
```suggestion
Built-in Exploration Components
```

File: doc/source/rllib/package_ref/utils.rst [hunk 5]
```diff
@@ -3,16 +3,201 @@
...
+
+   ~exploration.Exploration.get_state
+   ~exploration.Exploration.set_state
+
+
+
+Scheduler API
+-------------
+
+Scheduler are used to set scheduled values for varibles (in python, PyTorch or
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:71]:
```suggestion
Use a scheduler to set scheduled values for variables (in Python, PyTorch, or 
```

File: doc/source/rllib/package_ref/utils.rst [hunk 6]
```diff
@@ -3,16 +3,201 @@
...
+-------------
+
+Scheduler are used to set scheduled values for varibles (in python, PyTorch or 
+TensorFlow) based on a (int64) timestep input. The computed values are usually float32 
+types.
+
+
+
+
+Some Built-in Scheduler Components
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:78]:
```suggestion
Built-in scheduler components
```

File: doc/source/rllib/package_ref/utils.rst [hunk 7]
```diff
@@ -3,16 +3,203 @@
...
+
+Exploration is crucial in RL for enabling a learning agent to find new, potentially high-reward states by reaching unexplored areas of the environment.
+
+RLlib has several built-in exploration components that
+the different algorithms use. You can also customize an algorithm's exploration
+behavior by sub-classing the Exploration base class and implementing
+your own logic:
+
+Built-in exploration components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:19]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/source/rllib/package_ref/utils.rst [hunk 8]
```diff
@@ -3,16 +3,203 @@
...
+   :toctree: doc/
+
+   ~exploration.Exploration.before_compute_actions
+   ~exploration.Exploration.on_episode_start
+   ~exploration.Exploration.on_episode_end
+   ~exploration.Exploration.postprocess_trajectory
+
+
+Setting and getting the states
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:58]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/source/rllib/package_ref/utils.rst [hunk 9]
```diff
@@ -3,16 +3,203 @@
...
+
+Use a scheduler to set scheduled values for variables (in Python, PyTorch, or 
+TensorFlow) based on an (int64) timestep input. The computed values are usually float32 
+types.
+
+
+
+
+Built-in scheduler components
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:79]:
```suggestion
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

File: doc/source/rllib/package_ref/utils.rst [hunk 10]
```diff
@@ -3,16 +3,203 @@
...
+-------------
+
+Use a scheduler to set scheduled values for variables (in Python, PyTorch, or 
+TensorFlow) based on an (int64) timestep input. The computed values are usually float32 
+types.
+
+
+
+
+Built-in scheduler components
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:78]:
```suggestion
Built-in Scheduler components
```

Comment (angelinalg) [reply doc/source/rllib/package_ref/utils.rst:78]: Sorry, I didn't realize that Scheduler was the name of an object.

File: doc/source/rllib/package_ref/utils.rst [hunk 11]
```diff
@@ -3,16 +3,203 @@
...
+---------------
+
+Exploration is crucial in RL for enabling a learning agent to find new, potentially high-reward states by reaching unexplored areas of the environment.
+
+RLlib has several built-in exploration components that
+the different algorithms use. You can also customize an algorithm's exploration
+behavior by sub-classing the Exploration base class and implementing
+your own logic:
+
+Built-in exploration components
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:18]:
```suggestion
Built-in Exploration components
```

Comment (angelinalg) [reply doc/source/rllib/package_ref/utils.rst:18]: Sorry, I didn't realize that Exploration is an RLlib component. In that case, it should be capitalized.

File: doc/source/rllib/package_ref/utils.rst [hunk 12]
```diff
@@ -3,16 +3,203 @@
...
+
+.. autosummary::
+   :toctree: doc/
+
+   ~try_import_torch
+   ~try_import_tf
+   ~try_import_tfp
+
+
+Tensorflow Utilities
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:134]:
```suggestion
Tensorflow utilities
```

File: doc/source/rllib/package_ref/utils.rst [hunk 13]
```diff
@@ -3,16 +3,203 @@
...
+   ~make_tf_callable
+   ~minimize_and_clip
+   ~one_hot
+   ~reduce_mean_ignore_inf
+   ~scope_vars
+   ~warn_if_infinite_kl_divergence
+   ~zero_logps_from_actions
+
+
+Torch Utilities
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:157]:
```suggestion
Torch utilities
```

File: doc/source/rllib/package_ref/utils.rst [hunk 14]
```diff
@@ -3,16 +3,203 @@
...
+   ~one_hot
+   ~reduce_mean_ignore_inf
+   ~sequence_mask
+   ~warn_if_infinite_kl_divergence
+   ~set_torch_seed
+   ~softmax_cross_entropy_with_logits
+   ~Swish
+
+
+Numpy Utilities
```
Comment (angelinalg) [inline doc/source/rllib/package_ref/utils.rst:185]:
```suggestion
Numpy utilities
```

**Comment by angelinalg** [comment]
I made a best attempt to make the capitalization of titles consistent. Headings with ---- are title case and headings with ~~~~ are sentence case. If you feel strongly about all titles being title case, you can change them back. My advice is to be consistent with the rest of the docs.

**Review by gjoliver — APPROVED** [review]
